eclipse-che/che-dashboard
# Add editor upgrade check before starting a devworkspace

### What does this PR do?
Add editor upgrade check before starting a devworkspace.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/20193

### How to test it:
1. Remove all workspaces.
2. Open swagger for the dashboard.
3. Check templates list with swagger '{CHE}/dashboard/api/swagger'. It should be empty.
4. Create a new workspace from repo https://github.com/benoitf/use-reference-editor.
5. Check templates list. We should have one template.
This template should **not** have an annotation
```
      "che.eclipse.org/components-update-policy": "managed",
```
6. Create a new workspace from a sample. 

7. Check templates list. It should have one more template that includes
```
annotations": {
      "che.eclipse.org/components-update-policy": "managed",
      "che.eclipse.org/plugin-registry-url": "https://192.168.64.22.nip.io/plugin-registry..."
    }, 
```
8. On the UD tab open Networking and refresh tab. Filter request with 'template'.
There should be only one `get` request.

9. Modify existing template with the annotation `"che.eclipse.org/components-update-policy": "managed"`
using swagger. Simply execute `patch` request from `template` section using example payload:
```
[
  {
    "op": "replace",
    "path": "/spec/commands",
    "value": []
  }
]
```
10. On the UD tab open Networking and refresh tab. Filter request with 'template'.
There should be two requests(`get` and `patch`).

11. Refresh tab with UD one more time and filter request with 'template'.
It should be only one `get` request.
